packages/core/src/combinator/timeslice.js

```diff
@@ -37,77 +40,55 @@ class Since {
   }
 
   run (sink, scheduler) {
-    const min = new LowerBound(this.minSignal, sink, scheduler)
-    const max = new Bound(Infinity, sink)
-    const disposable = this.source.run(new TimeWindowSink(min, max, sink), scheduler)
-
-    return disposeAll([min, max, disposable])
-  }
-}
+    const min = new LowerBoundSink(this.minSignal, sink, scheduler)
+    const d = this.source.run(new SinceSink(min, sink), scheduler)
 
-class Bound extends Pipe {
-  constructor (value, sink) {
-    super(sink)
-    this.value = value
+    return disposeBoth(min, d)
   }
-
-  event () {}
-  end () {}
-
-  dispose () {}
 }
 
-class TimeWindowSink extends Pipe {
-  constructor (min, max, sink) {
+class SinceSink extends Pipe {
+  constructor (min, sink) {
     super(sink)
     this.min = min
-    this.max = max
   }
 
   event (t, x) {
-    if (t >= this.min.value && t < this.max.value) {
+    if (this.min.allow) {
       this.sink.event(t, x)
     }
   }
 }
 
-class LowerBound extends Pipe {
+class LowerBoundSink extends Pipe {
   constructor (signal, sink, scheduler) {
     super(sink)
-    this.value = Infinity
+    this.allow = false
     this.disposable = signal.run(this, scheduler)
   }
 
-  event (t /*, x */) {
-    if (t < this.value) {
-      this.value = t
-    }
+  event (/* t, x */) {
+    this.allow = true
+    this.dispose()
   }
 
   end () {}
 
   dispose () {
-    return this.disposable.dispose()
+    this.disposable.dispose()
   }
 }
 
-class UpperBound extends Pipe {
-  constructor (signal, sink, scheduler) {
+class UntilSink extends Pipe {
+  constructor (sink, disposable) {
     super(sink)
-    this.value = Infinity
-    this.disposable = signal.run(this, scheduler)
+    this.disposable = disposable
   }
 
   event (t, x) {
-    if (t < this.value) {
-      this.value = t
-      this.sink.end(t)
-    }
+    this.disposable.dispose()
```

Comment:
This explicit `dispose` is basically the crux of the fix, and is similar to what `take` does.

We may want to consider changing `run` to actually manage the final disposable in a similar way that runEffects does.  I think that would also have fixed this bug, but I felt like not fixing it here would leave `until` in a risky state.